### PR TITLE
fix: solve concurrency issues with MonitorResources

### PR DESCRIPTION
**Pull request by gerard-queralt**
This should, hopefully, fix that random crash on `ModuleResources::MonitorResources` when loading scenes. Since there's no 100% repro I can't be sure, though.

### CODE REVIEW

File: Source/FileSystem/ModuleResources.cpp [hunk 1]
```diff
@@ -563,70 +570,76 @@ void ModuleResources::MonitorResources()
...
-						toCreateMeta.push_back(resource);
-					}
-					// these type's assets are binary files changed in runtime
-					else if (resource->GetType() != ResourceType::Mesh && resource->GetType() != ResourceType::Material)
+					toCreateMeta.push_back(resource);
+				}
+				// these type's assets are binary files changed in runtime
+				else if (resource->GetType() != ResourceType::Mesh && resource->GetType() != ResourceType::Material)
+				{
+					long long assetTime = fileSystem->GetModificationDate(resource->GetAssetsPath().c_str());
```
Comment (MarcGonzalezMoratona) [inline Source/FileSystem/ModuleResources.cpp:610]:
```suggestion
					long long assetTimestamp = fileSystem->GetModificationDate(resource->GetAssetsPath().c_str());
```
not sure if needed, but I think this name is more accurate if we are looking for the exact date and time of the file modified

File: Source/FileSystem/ModuleResources.cpp [hunk 2]
```diff
@@ -563,70 +570,76 @@ void ModuleResources::MonitorResources()
...
+					long long libTime = fileSystem->GetModificationDate(
+						(resource->GetLibraryPath() + GENERAL_BINARY_EXTENSION).c_str());
+					if (assetTime > libTime)
```
Comment (MarcGonzalezMoratona) [inline Source/FileSystem/ModuleResources.cpp:613]:
```suggestion
					if (assetTimestamp > libTimestamp)
```

**Comment by MarcGonzalezMoratona** [comment]
I couldn't test it, but the code looks good. I added a few comments on the variable naming. No issues found. Don't forget to resolve conflicts 😉

**Review by MarcGonzalezMoratona — DISMISSED** [review]

**Comment by markmeliaa** [comment]
Was gonna approve this, but conflicts 👀

**Review by MarcGonzalezMoratona — APPROVED** [review]

**Review by Piterga12 — APPROVED** [review]
Tested it many times and doesn't crash anymore

**Pull request merged**